client.go

```diff
@@ -254,31 +254,25 @@ func (uc *Client) isEnabled(feature string, options ...FeatureOption) api.Strate
 		o(&opts)
 	}
 
-	var f *api.Feature
-	if opts.resolver != nil {
-		f = opts.resolver(feature)
-	} else {
-		f = uc.repository.getToggle(feature)
-	}
+	f := resolveToggle(uc, opts, feature)
 
 	ctx := uc.staticContext
 	if opts.ctx != nil {
 		ctx = ctx.Override(*opts.ctx)
 	}
 
 	if f == nil {
-		if opts.fallbackFunc != nil {
-			return api.StrategyResult{
-				Enabled: opts.fallbackFunc(feature, ctx),
-			}
-		} else if opts.fallback != nil {
+		return handleFallback(opts, feature, ctx)
+	}
+
+	if f.Dependencies != nil && len(*f.Dependencies) > 0 {
```

Comment:
I'm curious why you decided to do this check outside the isParentDependencySatisfied. I made the opposite decision but didn't have a strong opinion here

Reply:
To me it made more sense readability wise. If I don't have dependencies or the length of dependencies is 0 then I don't need to do check if `isParentDependencySatisfied(f, *ctx)`. Moving it inside that function obfuscates the check that determines whether or not that function should actually be run. I don't have strong opinions about this either, but I found it easier to reason about this way because now the function is limited to doing one thing and one thing only.